internal/auth0/connection/expand.go

```diff
@@ -773,3 +780,173 @@ func expandConnectionOptionsScopes(data *schema.ResourceData, options scoper) {
 		options.SetScopes(true, scope.(string))
 	}
 }
+
+func passThroughUnconfigurableConnectionOptions(
+	ctx context.Context,
+	api *management.Management,
+	connectionID string,
+	strategy string,
+	connection *management.Connection,
+) error {
+	var err error
+
+	switch strategy {
+	case management.ConnectionStrategyAD:
+		err = passThroughUnconfigurableConnectionOptionsAD(ctx, api, connectionID, connection)
+	case management.ConnectionStrategyAzureAD:
+		err = passThroughUnconfigurableConnectionOptionsAzureAD(ctx, api, connectionID, connection)
+	case management.ConnectionStrategySAML:
+		err = passThroughUnconfigurableConnectionOptionsSAML(ctx, api, connectionID, connection)
+	case management.ConnectionStrategyADFS:
+		err = passThroughUnconfigurableConnectionOptionsADFS(ctx, api, connectionID, connection)
+	case management.ConnectionStrategyPingFederate:
+		err = passThroughUnconfigurableConnectionOptionsPingFederate(ctx, api, connectionID, connection)
+	}
+
+	return err
+}
+
+func passThroughUnconfigurableConnectionOptionsAD(
+	ctx context.Context,
+	api *management.Management,
+	connectionID string,
+	connection *management.Connection,
+) error {
+	existingConnection, err := api.Connection.Read(ctx, connectionID)
+	if err != nil {
+		return err
+	}
+
+	existingOptions := existingConnection.Options.(*management.ConnectionOptionsAD)
```

Comment:
Can an already created enterprise connection, not have any options defined?

Reply:
I'm actually not sure and to your point, even if it were possible it should be incredibly rare. This would only be precautionary since we've seen occasional panics with type assertions. If you are certain that it is not applicable here, we can remove.

Reply:
Don't mind keeping the check for now.